modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogZoneTest.java

```diff
@@ -0,0 +1,464 @@
+
+        int ver = manager.latestCatalogVersion();
+        assertThat(manager.execute(renameZoneCmd), willCompleteSuccessfully());
+
+        assertEquals(ver + 1, manager.latestCatalogVersion());
+        assertEquals(TEST_ZONE_NAME, latestActiveCatalog().defaultZone().name());
+        assertEquals(defaultZone.id(), latestActiveCatalog().defaultZone().id());
+    }
+
+    @Test
+    public void testDefaultZone() {
+        CatalogZoneDescriptor defaultZone = latestActiveCatalog().defaultZone();
+
+        // Try to create zone with default zone name.
+        CatalogCommand cmd = CreateZoneCommand.builder()
+                .zoneName(defaultZone.name())
+                .partitions(42)
+                .replicas(15)
+                .storageProfilesParams(List.of(StorageProfileParams.builder().storageProfile(DEFAULT_STORAGE_PROFILE).build()))
+                .build();
+        assertThat(manager.execute(cmd), willThrow(DistributionZoneExistsValidationException.class));
+
+        // Validate default zone wasn't changed.
+        assertSame(defaultZone, manager.zone(defaultZone.name(), clock.nowLong()));
+    }
+
+    @Test
+    public void testAlterZone() {
+        String zoneName = TEST_ZONE_NAME;
+
+        CatalogCommand cmd = CreateZoneCommand.builder()
+                .zoneName(zoneName)
+                .partitions(42)
+                .replicas(15)
+                .dataNodesAutoAdjust(73)
+                .filter("expression")
+                .storageProfilesParams(List.of(StorageProfileParams.builder().storageProfile(DEFAULT_STORAGE_PROFILE).build()))
+                .build();
+
+        CatalogCommand alterCmd = AlterZoneCommand.builder()
+                .zoneName(zoneName)
+                .partitions(10)
+                .replicas(2)
+                .dataNodesAutoAdjustScaleUp(3)
+                .dataNodesAutoAdjustScaleDown(4)
+                .filter("newExpression")
+                .storageProfilesParams(List.of(StorageProfileParams.builder().storageProfile("test_profile").build()))
+                .build();
+
+        assertThat(manager.execute(cmd), willCompleteSuccessfully());
+        assertThat(manager.execute(alterCmd), willCompleteSuccessfully());
+
+        // Validate actual catalog
+        CatalogZoneDescriptor zone = manager.zone(zoneName, clock.nowLong());
+        assertNotNull(zone);
+        assertSame(zone, manager.zone(zone.id(), clock.nowLong()));
+
+        assertEquals(zoneName, zone.name());
+        assertEquals(10, zone.partitions());
+        assertEquals(2, zone.replicas());
+        assertEquals(INFINITE_TIMER_VALUE, zone.dataNodesAutoAdjust());
+        assertEquals(3, zone.dataNodesAutoAdjustScaleUp());
+        assertEquals(4, zone.dataNodesAutoAdjustScaleDown());
+        assertEquals("newExpression", zone.filter());
+        assertEquals("test_profile", zone.storageProfiles().profiles().get(0).storageProfile());
+    }
+
+    @Test
+    public void testCreateZoneWithSameName() {
```

Comment:
Let's also check if zone can be created after renaming/dropping the previous one with the same name.

Reply:
Thanks. Added